solr/core/src/test/org/apache/solr/security/BasicAuthStandaloneTest.java

```diff
@@ -96,35 +96,35 @@ public void testBasicAuth() throws Exception {
       String baseUrl = buildUrl(jetty.getLocalPort(), "/solr"); 
       httpSolrClient = getHttpSolrClient(baseUrl);
       
-      verifySecurityStatus(cl, baseUrl + authcPrefix, "/errorMessages", null, 20);
```

Comment:
Same comment here as for BasicAuthIntegrationTest - this test tries to prove that even with auth enabled you can do certain actions without creds.

Reply:
> you cannot anymore, which makes sense

With an explicit `blockUnknown=false` sure you can do anonymous requests if those are not covered by any of the permissions set up for authentication. So also this test class could be reverted to initial state and just explicitly set blockUnknown=false -- as long as blockUnknown=true is tested elsewhere.

Reply:
So my point was that there is no need to change the test behavior for this change, we can instead add the blockUnknown=false to these tests (which is exactly what users would do to keep existing behavior) and that's it.

I opened a PR https://github.com/MarcusSorealheis/lucene-solr/pull/1 with my proposal which removes a bunch of unnecessary and complicating changes. You should be able to merge it into this PR :)

Reply:
My latest simplifications reduces the patch from 1446 to 391 lines 😊